[Edsel-Tan] ip

Edsel-Tan opened the pull request:
# Makima 😳 
Chatbot with the following features
- CLI
- Bug-free (until bugs are found)

**Features:**
[X] Deleting and adding tasks
[X] Searching tasks
[ ] Actually useful features

# Installation steps
1. Download the jar file [here](https://github.com/Edsel-Tan/ip/tree/master)
2. Run java -jar Makima.jar

> [!IMPORTANT] 
> Tasks are saved in data/makima.txt. Do not delete it!

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello!");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Edsel-Tan committed:
removed Duke.java
added Makima.java

Edsel-Tan committed:
added user input functionality

Edsel-Tan committed:
added record and list functionality

Edsel-Tan committed:
added mark and unmark functionality

Edsel-Tan committed:
reworked implementation of tasks
added specific tasks

Edsel-Tan committed:
handled various user input related exceptions

Edsel-Tan committed:
added delete functionality

Edsel-Tan committed:
configured UI testing

Edsel-Tan committed:
Add FileManager class and constructors for Deadline, Event, ToDo

Edsel-Tan committed:
Fix missing data folder error

Edsel-Tan committed:
add save function

Edsel-Tan committed:
add getDate function

Edsel-Tan committed:
add DateTime functionality

Edsel-Tan committed:
ensure endtime is after starttime for events

Edsel-Tan committed:
fix event save error

Edsel-Tan committed:
add Parser class

Edsel-Tan committed:
organize into packages

Edsel-Tan committed:
Merge branch 'add-gradle-support'

Edsel-Tan committed:
add JUnit test for deadline

Edsel-Tan committed:
add JUnit test for event

Edsel-Tan committed:
add JavaDoc for Makima

Edsel-Tan committed:
add JavaDoc for FileManager

Edsel-Tan committed:
Edit function to conform to coding standard

Edsel-Tan committed:
Add find functionality

Edsel-Tan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/makima/command/Makima.java

Edsel-Tan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/makima/command/Makima.java

SherniceSng commented on src/main/java/makima/command/Makima.java:
```diff
@@ -0,0 +1,235 @@
+    public static final String LINE_SEPERATOR = "__________________";
+    public static final Scanner SC = new Scanner(System.in);
+    private boolean isRunning = true;
+    private boolean editedTasks;
```
Is it possible to rename this to sound more like a boolean?

SherniceSng commented on src/main/java/makima/io/FileManager.java:
```diff
@@ -0,0 +1,165 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class FileManager {
```
The use of noun for class name is good.

SherniceSng commented on src/main/java/makima/command/Makima.java:
```diff
@@ -0,0 +1,235 @@
+     * Prints a confirmation for completion of commands which
+     * edit the tasks array.
+     */
+    private void done() {
```
Is it possible to name this method with a word / phrase that is a more distinctive verb?

SherniceSng commented on src/main/java/makima/io/Parser.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Parser {
+
+    public static String getInput() {
```
Good use of verb for naming of method.

SherniceSng reviewed (commented):
LGTM. Most of the methods and classes follow the naming standard. Keep up the good work!

Edsel-Tan committed:
Add checkstyle support

Edsel-Tan committed:
Refactor code to fit java coding standard

Edsel-Tan committed:
Add gui resources

Edsel-Tan committed:
Add gui relevant classes

Edsel-Tan committed:
Restructure command parsing to use multiple command classes

Command parsing is done in a while loop.

This locks the thread which waits for user input and is not
compatible with the gui interface.

Extracting each commmand into a class allows us to model the chat
bot without the use of a for loop. It clarifies what input is
expected at each state of the bot.

Edsel-Tan committed:
Add shadowJar configuration

Edsel-Tan committed:
Assert valid index for task deletion

Edsel-Tan committed:
Assert valid user input during task creation

Edsel-Tan committed:
Add relevant javadoc

Edsel-Tan committed:
Change implementation of getTasksWithTerm(String) method

Method getTasksWithTerm uses for loop to iterate and search for
tasks.

Use of for loop is verbose and difficult to understand.

Let's switch to using streams to cut down and clarify what the code
is doing. Streams has many predefined functions which we can use to
cut down on code verbosity.

Edsel-Tan committed:
Merge pull request #1 from Edsel-Tan/branch-A-Assertions

Branch a assertions

Edsel-Tan committed:
Merge branch 'master' into branch-A-CodeQuality

Edsel-Tan committed:
Merge branch 'master' into branch-A-Streams

Edsel-Tan committed:
Merge branch 'master' into branch-A-Streams

Edsel-Tan committed:
Add new command prioritise

Edsel-Tan committed:
Change the appearence of high and low priority tasks

Edsel-Tan committed:
Add save functionality for priority levels

Edsel-Tan committed:
Fix bug with null state in prioritise command

Edsel-Tan committed:
Add return to menu function

Edsel-Tan committed:
Improve error messages

Edsel-Tan committed:
Reorganise load file functions

Load file is under a single method in file manager. Debugging and
exception handling is easier if we split the method into simpler
smaller methods.

Let's push common load functionalities into abstract task class and
let each subclass handle its relevant load functionalities.

Edsel-Tan committed:
Add file related exceptions

Edsel-Tan committed:
Add UI image

Edsel-Tan committed:
Update README

Edsel-Tan committed:
Update README

Edsel-Tan committed:
Update README

Edsel-Tan committed:
Add _config.yml

Edsel-Tan committed:
Add Jekyll

Edsel-Tan committed:
Update about.md

Edsel-Tan committed:
Revert "Update about.md"

This reverts commit 954870ccb4b4fe438ada0fc68b1846550b89cd84.

Edsel-Tan committed:
Revert "Add Jekyll"

This reverts commit 725db16242fd3f068f531e85c3707328506eb451.

Edsel-Tan committed:
Revert "Add _config.yml"

This reverts commit e65d760eb925b56178f051cfb68c7c49984bf43e.

Edsel-Tan committed:
Add command specifications

Edsel-Tan committed:
Update README.md

Edsel-Tan committed:
Merge branch 'master' of https://github.com/Edsel-Tan/ip

Edsel-Tan committed:
Update README.md

Edsel-Tan committed:
Update README.md

Edsel-Tan committed:
Merge branch 'master' of https://github.com/Edsel-Tan/ip

Edsel-Tan committed:
Fix bug with mark, unmark, prioritise, delete when task list is
empty

Edsel-Tan committed:
Update priority level to be case insensitive

Edsel-Tan committed:
Update README.md

Edsel-Tan committed:
Improve code quality

Edsel-Tan committed:
Fix application resizing
